senderWrapper.go

```diff
@@ -238,3 +230,21 @@ func undertaker(sw *CaduceusSenderWrapper) {
 		}
 	}
 }
+
+func createDeadlist(sw *CaduceusSenderWrapper, threshold time.Time) map[string]OutboundSender {
+	if sw == nil || threshold.IsZero() {
+		return nil
+	}
+
+	deadList := make(map[string]OutboundSender)
+	sw.mutex.Lock()
+	defer sw.mutex.RUnlock()
```

Comment:
This is mismatched.  `Lock` needs `Unlock`, and `RLock` needs `RUnlock`.  As written here, the `defer` will possibly panic since there was no corresponding acquisition of the write lock.  But, for sure this will deadlock since there is no code releasing the write lock acquired via `Lock`.